tck/faces50/phaseEvents/src/test/java/org/eclipse/ee4j/tck/faces/faces50/phaseEvents/Spec1443IT.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2024 Contributors to Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package org.eclipse.ee4j.tck.faces.faces50.phaseEvents;
+
+import static org.junit.Assert.assertEquals;
+
+import jakarta.faces.event.AfterPhase;
+import jakarta.faces.event.BeforePhase;
+
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import ee.jakarta.tck.faces.test.util.selenium.BaseITNG;
+import ee.jakarta.tck.faces.test.util.selenium.WebPage;
+
+public class Spec1443IT extends BaseITNG {
+
+    /**
+     * @see AfterPhase
+     * @see BeforePhase
+     * @see https://github.com/jakartaee/faces/issues/1443
+     */
+    @Test
+    public void testObservedPhases() {
+        WebPage page = getPage("spec1443.xhtml");
+        WebElement observedPhasesOnInitialRequest = page.findElement(By.id("observedPhases"));
+        assertEquals("""
+            [
+                beforePhaseDefault: RESTORE_VIEW 1,
+                beforeBeforePhaseAny: RESTORE_VIEW 1,
+                beforePhaseRestoreView: RESTORE_VIEW 1,
+                afterPhaseRestoreView: RESTORE_VIEW 1,
+                afterPhaseAny: RESTORE_VIEW 1,
+                afterPhaseDefault: RESTORE_VIEW 1,
+                beforePhaseDefault: RENDER_RESPONSE 6,
+                beforeBeforePhaseAny: RENDER_RESPONSE 6,
+                beforePhaseRenderResponse: RENDER_RESPONSE 6
```

Comment:
NOTE: the afterPhase for RENDER_RESPONSE can indeed not be asserted because of it being fired after render response ... if we also want to assert it then we need to adjust the test with some robust async update in current view scope retriving the events fired in previous request